Tools/autotest/ardusub.py

```diff
@@ -1133,6 +1160,150 @@ def SurfaceSensorless(self):
         self.disarm_vehicle()
         self.context_pop()
 
+    def wait_for_distance(self, distance, timeout=100):
+        original_location = self.mav.recv_match(type='GLOBAL_POSITION_INT', blocking=True)
```

Comment:
avoid recv_match in new code, use assert_receive_message.  We also have a helper to get a location object somewhere...